### PR TITLE
Helper function to access billing api and update the capped amount for a usage billing plan

### PR DESCRIPTION
**Pull request by andy-liuu**
### WHY are these changes introduced?

(https://github.com/Shopify/first-party-library-planning/issues/669)

### WHAT is this pull request doing?

A helper function to access the billing API and allow apps to update the max charge (terms and conditions) of a usage billing plan

## Type of change

- [ ] Patch: Bug (non-breaking change which fixes an issue)
- [x] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have used `pnpm changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [ ] I have added/updated tests for this change
- [ ] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

**Comment by andy-liuu** [comment]
todo: changeset after self review

**Review by Ydoug36 — APPROVED** [review]

**Comment by andy-liuu** [comment]
May need to add 'Skip Markdown Link Check' due to some untouched links becoming broken recently. But I'll still request a review and wait a bit to see if this resolves itself.

File: packages/apps/shopify-api/lib/__test-helpers__/graphql-base-request.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {LATEST_API_VERSION} from '../types';
```
Comment (lizkenyon) [inline packages/apps/shopify-api/lib/__test-helpers__/graphql-base-request.ts:1]: Nice refactor here!

File: packages/apps/shopify-api/docs/reference/billing/update-usage-capped-amount.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### cappedAmount.currencyCode
+
+`string` | :exclamation: required
+
+The currency code for the capped amount.
+
+## Returns
+
+`userErrors: string[]`
```
Comment (lizkenyon) [inline packages/apps/shopify-api/docs/reference/billing/update-usage-capped-amount.md:51]: For the return we will either throw an error with the userErrors, if they exists _or_ we will return an object with confirmationUrl and appSubscription. We won't ever have both.

File: packages/apps/shopify-api/lib/billing/types.ts [hunk 3]
```diff
@@ -450,6 +450,12 @@ export interface AppPlanDiscount {
   value: AppPlanDiscountAmount;
 }
 
+export interface AppSubscriptionLineItemUpdatePayload {
```
Comment (lizkenyon) [inline packages/apps/shopify-api/lib/billing/types.ts:453]: We will need to create an additional type like `AppSubscriptionLineItemUpdateResponse` that is just the `confirmationUrl` and the `appSubscription`.

Comment (lizkenyon) [reply packages/apps/shopify-api/lib/billing/types.ts:453]: Similar to how the cancel [method returns an AppSubscription](https://github.com/Shopify/shopify-app-js/blob/6abecca30bce38f140fbaf31f0137d4a30cf1561/packages/apps/shopify-api/lib/billing/types.ts#L627)

And not a CancelResponse.

File: packages/apps/shopify-api/lib/billing/update-usage-subscription-capped-amount.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  }
+}
+`;
+
+export function updateUsageCappedAmount(
+  config: ConfigInterface,
+): BillingUpdateUsageCappedAmount {
+  return async function updateUsageCappedAmount(
+    params: BillingUpdateUsageCappedAmountParams,
+  ): Promise<AppSubscriptionLineItemUpdatePayload> {
```
Comment (lizkenyon) [inline packages/apps/shopify-api/lib/billing/update-usage-subscription-capped-amount.ts:52]: This will return the new type.

**Review by lizkenyon — COMMENTED** [review]
Looks good! Just a small update required! ⭐

**Review by lizkenyon — APPROVED** [review]
Looks good! ⭐

**Pull request merged**